lib/client/mfa/cli.go

```diff
@@ -35,117 +38,155 @@ import (
+		availableMethods = append(availableMethods, CLIMFATypeWebauthn)
+	}
+	if promptSSO {
+		availableMethods = append(availableMethods, CLIMFATypeSSO)
+	}
+	if promptOTP {
+		availableMethods = append(availableMethods, CLIMFATypeOTP)
+	}
 
-		// Fire TOTP goroutine.
-		var otpCancelAndWait func()
-		if runOpts.PromptTOTP {
-			otpCtx, otpCancel := context.WithCancel(ctx)
-			otpDone := make(chan struct{})
-			otpCancelAndWait = func() {
-				otpCancel()
-				<-otpDone
-			}
+	// Check off unsupported methods.
+	if promptWebauthn && !c.WebauthnSupported {
+		promptWebauthn = false
+		slog.DebugContext(ctx, "hardware device MFA not supported by your platform")
+	}
 
-			wg.Add(1)
-			go func() {
-				defer func() {
-					wg.Done()
-					otpCancel()
-					close(otpDone)
-				}()
-
-				quiet := c.cfg.Quiet || dualPrompt
-				resp, err := c.promptTOTP(otpCtx, quiet)
-				respC <- MFAGoroutineResponse{Resp: resp, Err: trace.Wrap(err, "TOTP authentication failed")}
-			}()
-		}
+	// Prefer whatever method is requested by the client.
+	var chosenMethod string
+	switch {
+	case c.PreferSSO && promptSSO:
+		chosenMethod = CLIMFATypeSSO
+		promptWebauthn, promptOTP = false, false
+	case c.PreferOTP && promptOTP:
+		chosenMethod = CLIMFATypeOTP
+		promptWebauthn, promptSSO = false, false
+	}
 
-		// Fire Webauthn goroutine.
-		if runOpts.PromptWebauthn {
-			wg.Add(1)
-			go func() {
-				defer func() {
-					wg.Done()
-					// Important for dual-prompt, harmless otherwise.
-					webauthnwin.ResetPromptPlatformMessage()
-				}()
-
-				// Get webauthn prompt and wrap with otp context handler.
-				prompt := &webauthnPromptWithOTP{
-					LoginPrompt:      c.getWebauthnPrompt(ctx, dualPrompt),
-					otpCancelAndWait: otpCancelAndWait,
-				}
-
-				resp, err := c.promptWebauthn(ctx, chal, prompt)
-				respC <- MFAGoroutineResponse{Resp: resp, Err: trace.Wrap(err, "Webauthn authentication failed")}
-			}()
+	// Use stronger auth methods if hijack is not allowed.
+	if !c.AllowStdinHijack && (promptWebauthn || promptSSO) {
+		promptOTP = false
+	}
+
+	// If we have multiple viable options, prefer Webauthn > SSO > OTP.
+	switch {
+	case promptWebauthn:
+		chosenMethod = CLIMFATypeWebauthn
+		promptSSO = false
+
+		// If a specific webauthn attachment was requested, skip OTP.
+		// Otherwise, allow dual prompt with OTP.
+		if c.AuthenticatorAttachment != wancli.AttachmentAuto {
+			promptOTP = false
+		} else if promptOTP {
+			chosenMethod = fmt.Sprintf("%v and %v", CLIMFATypeWebauthn, CLIMFATypeOTP)
 		}
+	case promptSSO:
+		chosenMethod = CLIMFATypeSSO
+		promptOTP = false
+	case promptOTP:
+		chosenMethod = CLIMFATypeOTP
 	}
 
-	return HandleMFAPromptGoroutines(ctx, spawnGoroutines)
+	if chosenMethod != "" {
+		fmt.Fprintf(c.Writer, "Available MFA methods [%v]. Continuing with %v.\n", strings.Join(availableMethods, ", "), chosenMethod)
+		fmt.Fprintf(c.Writer, "If you wish to perform MFA with another method, specify with flag --mfa-mode=<sso,otp>.\n\n")
```

Comment:
Suggestion: Do a single printf?

```suggestion
		const msg = "" +
				"Available MFA methods [%v]. Continuing with %v.\n" +
				"If you wish to perform MFA with another method, specify with flag --mfa-mode=<sso,otp>.\n\n"
		fmt.Fprintf(c.Writer, msg, strings.Join(availableMethods, ", "), chosenMethod)
```